### PR TITLE
labels: Extra tests and comment for WithoutLabels method.

### PR DESCRIPTION
**Pull request by bwplotka**
Implementation was not straightforward so helping next readers. (: 

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: pkg/labels/labels.go [hunk 1]
```diff
@@ -275,6 +275,7 @@ func (ls Labels) WithoutEmpty() Labels {
 		if v.Value != "" {
 			continue
 		}
+		// Copy slice without empty labels only if needed.
```
Comment (brian-brazil) [inline pkg/labels/labels.go:278]: I don't understand what this comment is trying to say.

Comment (bwplotka) [reply pkg/labels/labels.go:278]: That we only touch the slice and copy things between if at least single label is empty (: Will rework comment

Comment (bwplotka) [reply pkg/labels/labels.go:278]: reworded!

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**